app/validators/array_type_validator.rb

```diff
@@ -1,5 +1,5 @@
 class ArrayTypeValidator < ActiveModel::EachValidator
   def validate_each(record, attribute, value)
-    record.errors.add attribute, _("must be comma separated") unless Array(value).map { |item| item.to_s.match /\A\S+\z/ }.all?
+    record.errors.add attribute, _("must be comma separated") unless Array(value).map { |item| item.to_s.strip.match /\A\S+\z/ }.all?
```

Comment:
Unrelated to this PR, but I think `.all?` accepts a block and this could be optimized:
```suggestion
    record.errors.add attribute, _("must be comma separated") unless Array(value).all? { |item| item.to_s.strip.match?(/\A\S+\z/) }
```

Note I didn't test this and it's probably better to do in a separate PR

Reply:
> To be explicit: the changes to the validators should be reverted. They should not accept any whitespace.

@ekohl Are you suggesting that we shall apply the change somewhere else or we can modify the Regex for the same validation?

Reply:
I'd apply the change elsewhere. Normally the flow is:

* User input
* Apply normalization
* Apply validators
* Process data

Processing data can mean storing it in the DB, but you want it to be as clean as possible. The main reason is that if you can assume that data in the database doesn't have whitespace around the value, you can keep the code that uses data from the database simpler. Or put in another way: if you must remember to always call `.strip` on a value, you can easily introduce bugs. That's more important than the potential performance difference, but that's another factor.

An example:
```ruby
if model.value == 'value'
  # Do X
else
  # Do Y
end
```

This will work most of the time, but if the user (accidentally) has spaces in the value it can introduce a bug.

Looking back at the flow, this means we want to clean up the data before processing it. A validator is then only supposed to validate. It never changes data. This means normalization can happen in the UI, before it's submitted, or in the controller before it's passed to the validators. But the validator must remain strict.

Reply:
Thanks a lot for the clarification Ewoud!!